happytransformer/happy_word_prediction.py

```diff
@@ -15,36 +13,56 @@
 
 
 class HappyWordPrediction(HappyTransformer):
-    def __init__(self, model_type="BERT",
-                 model_name="bert-large-uncased-whole-word-masking-finetuned-squad", device=None):
+    """
+    A user facing class for text classification
+    """
+    def __init__(self, model_type="DISTILBERT",
+                 model_name="distilbert-base-uncased"):
         model = None
         tokenizer = None
 
         if model_type == "BERT":
             model = BertForMaskedLM.from_pretrained(model_name)
             tokenizer = BertTokenizerFast.from_pretrained(model_name)
 
-        elif model_type == "ROBERTA":
-            model = RobertaForMaskedLM.from_pretrained(model_name)
-            tokenizer = RobertaTokenizerFast.from_pretrained(model_name)
+        elif model_type == "DISTILBERT":
+            model = DistilBertForMaskedLM.from_pretrained(model_name)
+            tokenizer = DistilBertTokenizerFast.from_pretrained(model_name)
 
-        super().__init__(model_type, model_name, model, tokenizer, device)
+        super().__init__(model_type, model_name, model, tokenizer)
         device_number = 1 if torch.cuda.is_available() else -1
-        self._pipeline = FillMaskPipeline(model=model,
-                                                    tokenizer=tokenizer, device=device_number)
-        self._trainer = MWPTrainer(model,
-                                  model_type, tokenizer, self._device, self.logger)
-    def predict_masks(self):
-        raise NotImplementedError()
+        self._pipeline = FillMaskPipeline(model=model, tokenizer=tokenizer, device=device_number)
+        self._trainer = MWPTrainer(model, model_type, tokenizer, self._device, self.logger)
+
+    def predict_mask(self, text, targets=None, top_k=1):
+        """
+        :param text: A string that contains the model's mask token
+        :param targets: Optional. A list of strings of potential answers.
+        All other answers will be ignored
+        :param top_k: number of results. Default is 1
+        :return: If top_k ==1: a dictionary with the keys "score" and "token_str"
+                if  top_k >1: a list of dictionaries described above in order by score
+        """
+        if isinstance(text, list):
+            raise ValueError("the \"text\" argument must be a single string")
 
-    def predict_mask(self):
-        raise NotImplementedError()
+        result = self._pipeline(text, targets=targets, top_k=top_k)
+        if top_k == 1:
+            result = result[0]
+            del result['sequence']
+            del result['token']
+        else:
+            for answer in result:
+                del answer['sequence']
+                del answer['token']
+        return result
```

Comment:
returning a dict with what we want explicitly would be cleaner than deleting things